src/client/nativeCommunication.ts

```diff
@@ -108,7 +111,9 @@ function launchSlideshow(src: string | null): void {
             const url = updateUrl(image?.currentSrc ?? image.src);
             const caption =  image.getAttribute('data-caption') ?? undefined;
             const credit = image.getAttribute('data-credit') ?? undefined;
-            return new Image({ url, caption, credit });
+            const width = getImageWidth(url);
+            const height = width * parseFloat(image.getAttribute('data-ratio') ?? '0.56');
```

Comment:
It's possible height could be `NaN` here. What happens in the Thrift transport layer/on the native side if a number value is `NaN`?

Reply:
Could it? `getImageWidth` returns a number and `image.width` `image.height` aren't optional

Reply:
`parseFloat` can return a `NaN`, which could happen, for example, if `data-ratio` gets mangled.

Reply:
ok we might have to avoid showing the image without having a height/width.